R/DBConnection.R

```diff
@@ -106,6 +106,41 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' @export
+dbBreak <- structure(list(), class = "dbiAbort")
```

Comment:
With your implementation of `dbBreak`, you'd have to say `stop(dbBreak)` to get out of the loop. I think `dbBreak()` is better for hiding the implementation details, which should not be the user's concern. We also may allow a custom error message in `dbBreak()`.